x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_table.test.tsx

```diff
@@ -32,113 +31,112 @@ describe('EnginesTable', () => {
     } as EngineDetails,
   ];
   const pagination = {
-    totalEngines: 50,
     pageIndex: 0,
-    onPaginate,
+    pageSize: 10,
+    totalItemCount: 50,
+    hidePerPageOptions: true,
   };
   const props = {
-    data,
+    items: data,
+    loading: false,
     pagination,
+    onChange,
   };
 
-  const wrapper = mountWithIntl(<EnginesTable {...props} />);
-  const table = wrapper.find(EuiBasicTable);
+  describe('basic table', () => {
+    const wrapper = mountWithIntl(<EnginesTable {...props} />);
+    const table = wrapper.find(EuiBasicTable);
 
-  it('renders', () => {
-    expect(table).toHaveLength(1);
-    expect(table.prop('pagination').totalItemCount).toEqual(50);
+    it('renders', () => {
+      expect(table).toHaveLength(1);
+      expect(table.prop('pagination').totalItemCount).toEqual(50);
 
-    const tableContent = table.text();
-    expect(tableContent).toContain('test-engine');
-    expect(tableContent).toContain('Jan 1, 1970');
-    expect(tableContent).toContain('English');
-    expect(tableContent).toContain('99,999');
-    expect(tableContent).toContain('10');
+      const tableContent = table.text();
+      expect(tableContent).toContain('test-engine');
+      expect(tableContent).toContain('Jan 1, 1970');
+      expect(tableContent).toContain('English');
+      expect(tableContent).toContain('99,999');
+      expect(tableContent).toContain('10');
 
-    expect(table.find(EuiPagination).find(EuiButtonEmpty)).toHaveLength(5); // Should display 5 pages at 10 engines per page
-  });
+      expect(table.find(EuiPagination).find(EuiButtonEmpty)).toHaveLength(5); // Should display 5 pages at 10 engines per page
+    });
 
-  it('contains engine links which send telemetry', () => {
-    const engineLinks = wrapper.find(EuiLinkTo);
+    it('contains engine links which send telemetry', () => {
+      const engineLinks = wrapper.find(EuiLinkTo);
 
-    engineLinks.forEach((link) => {
-      expect(link.prop('to')).toEqual('/engines/test-engine');
-      link.simulate('click');
+      engineLinks.forEach((link) => {
+        expect(link.prop('to')).toEqual('/engines/test-engine');
+        link.simulate('click');
 
-      expect(mockTelemetryActions.sendAppSearchTelemetry).toHaveBeenCalledWith({
-        action: 'clicked',
-        metric: 'engine_table_link',
+        expect(mockTelemetryActions.sendAppSearchTelemetry).toHaveBeenCalledWith({
+          action: 'clicked',
+          metric: 'engine_table_link',
+        });
       });
     });
-  });
 
-  it('triggers onPaginate', () => {
-    table.prop('onChange')({ page: { index: 4 } });
-
-    expect(onPaginate).toHaveBeenCalledWith(5);
+    it('triggers onPaginate', () => {
+      table.prop('onChange')({ page: { index: 4 } });
+      expect(onChange).toHaveBeenCalledWith({ page: { index: 4 } });
+    });
   });
 
-  it('handles empty data', () => {
-    const emptyWrapper = mountWithIntl(
```

Comment:
I deleted the `empty data` test as it didn't really seem useful/to apply any longer with the new pagination helpers